scripts/UpdateBuildTools.ps1

```diff
@@ -48,49 +48,57 @@ function Get-KoreBuildVersion {
 
-    # Executes a command that no-ops. The only thing we really need is the updated version of korebuild-lock.txt
-    & "$RepoRoot/run.ps1" -Update --help | Out-Null
+        # Executes a command that no-ops. The only thing we really need is the updated version of korebuild-lock.txt
+        & "$repoRoot/run.ps1" -Update --help | Out-Null
 
-    $newVersion = Get-KoreBuildVersion
+        $newVersion = Get-KoreBuildVersion
 
-    if ($oldVersion -eq $newVersion) {
-        Write-Host -ForegroundColor Magenta 'No changes to build tools'
-        exit 0
-    }
+        if ($oldVersion -eq $newVersion) {
+            Write-Host -ForegroundColor Magenta 'No changes to build tools'
+            exit 0
+        }
 
-    Invoke-Block { git add "$RepoRoot/korebuild-lock.txt" }
-    Invoke-Block { git add "$RepoRoot/build/dependencies.props" }
+        Invoke-Block { git add "$repoRoot/korebuild-lock.txt" }
+        Invoke-Block { git add "$repoRoot/build/dependencies.props" }
 
-    $shortMessage = "Updating BuildTools from $oldVersion to $newVersion"
-    # add this to the commit message to make it possible to filter commit triggers based on message
-    $message = "$shortMessage`n`n[auto-updated: buildtools]"
+        $shortMessage = "Updating BuildTools from $oldVersion to $newVersion"
+        # add this to the commit message to make it possible to filter commit triggers based on message
+        $message = "$shortMessage`n`n[auto-updated: buildtools]"
 
-    if (-not $NoCommit -and ($Force -or ($PSCmdlet.ShouldContinue($shortMessage, 'Create a new commit with these changes?')))) {
+        if (-not $NoCommit -and ($Force -or ($PSCmdlet.ShouldContinue($shortMessage, 'Create a new commit with these changes?')))) {
 
-        $gitConfigArgs = @()
-        if ($GitAuthorName) {
-            $gitConfigArgs += '-c', "user.name=$GitAuthorName"
-        }
+            $gitConfigArgs = @()
+            if ($GitAuthorName) {
+                $gitConfigArgs += '-c', "user.name=$GitAuthorName"
+            }
 
-        if ($GitAuthorEmail) {
-            $gitConfigArgs += '-c', "user.email=$GitAuthorEmail"
-        }
+            if ($GitAuthorEmail) {
+                $gitConfigArgs += '-c', "user.email=$GitAuthorEmail"
+            }
 
-        Invoke-Block { git @gitConfigArgs commit -m $message @GitCommitArgs }
-    }
-    else {
-        # If composing this script with others, return the message that would have been used
-        return @{
-            message = $message
+            Invoke-Block { git @gitConfigArgs commit -m $message @GitCommitArgs }
+        }
+        else {
+            # If composing this script with others, return the message that would have been used
+            return @{
+                message = $message
+            }
         }
     }
+    finally {
+        Pop-Location
+    }
 }
-finally {
-    Pop-Location
+
+$submodules = Get-Submodules $RepoRoot
+foreach ($submodule in $submodules) {
```

Comment:
How do we expect this change to be persisted? Doesn't this already happen as a part of UpdateRepos?

Reply:
So it is!